### PR TITLE
Organize Sphinx docs into separate User and Developer guide sections

### LINKED ISSUE #341: Organize sphinx docs
All of the pages on our sphinx docs are at the same level. A reasonable hierarchy (User guide, Dev guide, Theory guide, etc) would be helpful for navigating these pages.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
View at https://serac.readthedocs.io/en/docs-essman-organize/

Resolves #341

### CODE REVIEW

File: src/docs/index.rst [hunk 1]
```diff
@@ -249,11 +249,5 @@ LLNL-CODE-805541
...
 
-   sphinx/developer_guide
-   sphinx/style_guide
-   sphinx/logging
-   sphinx/expr_templates
-   sphinx/memory_checking
-   sphinx/profiling
-   sphinx/docker_info
-   sphinx/input_schema
+   sphinx/user_guide/index
```
Comment (jamiebramwell) [inline src/docs/index.rst:35]: We should probably change the code reference to Smith at the top of this file. Also, do we want the build instructions to be on a separate page?

File: src/docs/sphinx/dev_guide/index.rst [hunk 2]
```diff
@@ -39,26 +26,26 @@ Developers have two workflows for creating new physics modules:
...
-* `EquationSolver <../doxygen/html/classserac_1_1EquationSolver.html>`_: Class for solving nonlinear and linear systems of equations.
-* `FiniteElementState <../doxygen/html/classserac_1_1FiniteElementState.html>`_: Data structure describing a solution field and its underlying finite element discretization.
+* `BasePhysics <../../doxygen/html/classserac_1_1BasePhysics.html>`_: Interface class for a generic PDE simulation module.
+* `BoundaryCondition <../../doxygen/html/classserac_1_1BoundaryCondition.html>`_: Class for storage of boundary condition-related data.
+* `BoundaryConditionManager <../../doxygen/html/classserac_1_1BoundaryConditionManager.html>`_: Storage class for related boundary conditions.
+* `EquationSolver <../../doxygen/html/classserac_1_1mfem__ext_1_1EquationSolver.html>`_: Class for solving nonlinear and linear systems of equations.
+* `FiniteElementState <../../doxygen/html/classserac_1_1FiniteElementState.html>`_: Data structure describing a solution field and its underlying finite element discretization.
 
 Frequently Used Modern C++ Features
```
Comment (jamiebramwell) [inline src/docs/sphinx/dev_guide/index.rst]: This should probably be a separate page.

**Review by jamiebramwell — APPROVED** [review]

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

File: src/docs/index.rst (hunk 1 above)
Comment (joshessman-llnl) [reply src/docs/index.rst:35]: I think having the build instructions/quickstart as a separate page makes sense

**Pull request merged**